clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleApplicationProvider.java

```diff
@@ -66,32 +68,50 @@
     return data.stream().map(this::applicationFromCacheData).collect(toSet());
   }
 
-  @Override
-  public Application getApplication(String name) {
+  @Value
+  static class ApplicationCacheData {
+    Map<String, Object> applicationAttributes;
+    Set<String> clusterIdentifiers;
+    Set<String> instanceIdentifiers;
+  }
 
+  ApplicationCacheData getApplicationCacheData(String name) {
     CacheData cacheData =
         cacheView.get(
             APPLICATIONS.getNs(),
             Keys.getApplicationKey(name),
             RelationshipCacheFilter.include(CLUSTERS.getNs(), INSTANCES.getNs()));
-    if (cacheData == null) {
-      return null;
-    }
+    return getApplicationCacheData(cacheData);
+  }
 
-    return applicationFromCacheData(cacheData);
+  ApplicationCacheData getApplicationCacheData(CacheData cacheData) {
```

Comment:
I think this can be private? go/java-practices/visibility

Reply:
Thanks Plumpy good call :pray: :pray: